### ISSUE TITLE
PageFactory implementation for By/AppiumBy object types

### ISSUE THREAD

**Issue by npopov8** [appium/java-client#1940]
## Description

TL;DR, I would like to know if there will be demand/usage for such an implementation. I have ready code if there is interest, the details are shared below.

## Environment

* Java client build version or git revision if you use some snapshot: `8.5.0`
* Appium server version or git revision if you use some snapshot: `v2.0.0-beta.71`

## Details

I recently encountered the following struggle:

I was re-building a cross-platform testing solution and was about to implement changes with PageFactory, as it’s a very efficient way to build tests fast and with less code for 2 platforms at the same time. What happened is that one of the internal SDK’s, used by the project in question, got updated to Java 17 and created a compatibility issue with PageFactory (the problem is that PageFactory uses cglib, which is not maintained nor updated for Java17).

After some thinking I went ahead and implemented a custom and very simplified implementation of a custom PageFactory, which works with By, instead of WebElement. The reasons I did it like that are:

- Simplicity - this way the code is quite simplified, without having to instantiate the actual WebElements right away and thus no need for complicated proxying of the objects.
- Flexibility - WebElements are great but they are not as flexible as the By locators, and also can be acquired at will as long as you have the By for the given element, but not the other way around.

This ended up being quite effective and I was wondering if this is something that may be of interest/demand? I can contribute my code if moderators agree this could be useful for many many people, I just want to know if this is something that the Appium community may be interested in. It’s purpose will not only be a (more stable) alternative for PageFactory, but also maybe a successor for it, unless the Selenium community updates that part of Selenium to not depend on cglib.

**Comment by npopov8** [comment]
@mykola-mokhnach do you think this is something that could be beneficial or should I close this issue?

**Comment by itkhanz** [comment]
I like your idea. Would be nice to see some code example of how you implemented it.

I also use PageFactory for automation of cross platform apps and once I had been stuck due to final nature of Java annotations not allowing me to use dynamic locators with pagefactory annotations so I had to use a custom implementation to build cross platform By locators where I can inject dynamic value.
